tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Test_SchwarzAlgorithm.yaml

```diff
@@ -0,0 +1,61 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# The test problem being solved here is a DG-discretized 1D Poisson equation
+# -u''(x) = f(x) on the interval [0, pi] with source f(x)=sin(x) and homogeneous
+# Dirichlet boundary conditions such that the solution is u(x)=sin(x) as well.
+#
+# Details:
+# - Domain decomposition: 2 elements with 3 LGL grid-points each
+# - "Primal" DG formulation (no auxiliary variable)
+# - Not multiplied by mass matrix so the operator is not symmetric
+# - Mass-lumping: inverse mass matrix is approximated by diagonal
+# - Internal penalty flux with sigma = 1.5 * (N_points - 1)^2 / h
+
+DomainCreator:
+  Interval:
+    LowerBound: [0]
+    UpperBound: [3.141592653589793]
+    IsPeriodicIn: [false]
+    InitialRefinement: [1]
+    InitialGridPoints: [3]
+
+LinearOperator:
```

Comment:
probably not for this PR, because this is in a lot of the linear solver tests, but I think encoding the explicit matrices in the input file seems pretty difficult to maintain or update. Instead, it would be preferable for the input file to encode the meaningful parameters you used to generate these numbers, and the code you used to generate them be ported into the constructor for the linear operator (I'm guessing you probably wrote some python code to make these matrices...)

Reply:
I'm happy to discuss this some time. My argument is: The linear solvers should be (and are) problem-agnostic, i.e. they don't care how the linear operator is built. It could be some explicit matrix or some FD, spectral or DG operator, it shouldn't matter. I don't want these tests to depend on the whole DG code, since that's not what they are supposed to test, so I only feed them some explicit matrix and its solution. I chose matrices that represent DG problems only so they are close to what we're actually going to use the solvers for.

Reply:
I'm fine with the linear solvers being problem agnostic, but I'm of the firm opinion that using precise numerical values in the input file indicates that there should be an additional specialization somewhere in the c++. Here, I'm saying you should have some separate factory that can create a linear operator that has the 'interesting' parameters, rather than a bunch of doubles.
No one is going to be able to create this type of input without a separate piece of software for writing out the matrix, so my reasoning is that taking explicit numerics in the input file is roughly equivalent to introducing an undocumented dependency, with a very buggy import method (humans moving the data around manually).

Reply:
I agree that humans can be very buggy :D Perhaps I should just use a simple FD matrix instead for these tests, instead of a DG operator. The DG operator takes a lot of Python code to build, but a FD matrix could be found in any book...

Reply:
Perhaps the FD operator would be acceptable, if you don't think that would degrade the rigor of the tests at all. The DG operator seems to have the advantage of being more similar to systems that will be used in practice. I still think that the eventual goal (not for this PR -- the complaint is more general and if addressed should be tackled separately) should be to use inheritance, or the factory options technique to create specializations of the linear operator associated with specific problems, so that the generating code can appear in test helpers.